demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+                                   / cv::getTickFrequency());
+                }
+                avgLatency = latencySum / framesNum;
+                accuracy = static_cast<double>(rightPredictionsCount) / totalPredictionsCount;
+                gridMat.textUpdate(avgFPS, avgLatency, accuracy, isTestMode, hasClasses, presenter);
+            }
+            else if (!emptyInferRequests.empty()) {
+                mutex.unlock();
+                cv::Mat nextImage = inputImages[nextImageIndex];
+                long long IRstartTime = cv::getTickCount();
+                resizeImage(nextImage, modelInputResolution);
+                emptyInferRequests.front().images.push_back(
+                    IRInfo::IRImage(nextImage,
+                                    std::stoul(hasClasses ? classIndexes[nextImageIndex] : "0"),
+                                    IRstartTime));
+                nextImageIndex++;
+                if (nextImageIndex == inputImagesCount) {
+                    nextImageIndex = 0;
+                }
+                if (emptyInferRequests.front().images.size() == FLAGS_b) {
+                    auto emptyIR = emptyInferRequests.front();
+                    auto& ir = emptyIR.ir;
+                    auto irImages = emptyIR.images;
+
+                    emptyIR.ir.SetCompletionCallback([&ir,
+                                                      irImages,
+                                                      &completedInferRequests,
+                                                      &mutex,
+                                                      &condVar,
+                                                      &irCallbackException] {
+                        try {
+                            {
+                                std::lock_guard<std::mutex> lock(mutex);
+
+                                completedInferRequests.push(IRInfo(ir, irImages));
+                            }
+                            condVar.notify_one();
+                        }
+                        catch(...) {
+                            irCallbackException = std::current_exception();
+                        }
+                    });
+                    
+                    auto inputBlob = emptyIR.ir.GetBlob(inputBlobName);
+                    for (unsigned i = 0; i < FLAGS_b; i++) {
+                        matU8ToBlob<uint8_t>(emptyIR.images[i].mat, inputBlob, i);
+                    }
+                    emptyIR.ir.StartAsync();
+                    emptyInferRequests.pop();
+                }
+            }
+            
+            mutex.lock();
+            while (emptyInferRequests.empty() && completedInferRequests.empty()) {   
+                condVar.wait(lock);
+            }
+            mutex.unlock();
+
+            currentTime = std::chrono::system_clock::now();
+            elapsedSeconds = currentTime - startTime;
+        } while (27 != key && (FLAGS_time == -1 || elapsedSeconds.count() < FLAGS_time));
```

Comment:
The pressed key should be passed to `presenter` to allow toggling utilization graphs.

Reply:
In current implementation presenter will receive a key only if `emptyInferRequests.empty() && completedInferRequests.empty()`.
exception presence check is also performed only if `emptyInferRequests.empty() && completedInferRequests.empty()`.
It should be done more frequently. For example if there was a large `-nireq`, `emptyInferRequests` will never be empty.

`Q` and `q` values can also be added to behave like 27.

Reply:
Exception presence check is still performed only if `emptyInferRequests.empty() && completedInferRequests.empty()`.